core/extensions.rs

```diff
@@ -31,6 +31,7 @@ pub enum ExtensionFileSourceCode {
 pub struct ExtensionFileSource {
   pub specifier: &'static str,
   pub code: ExtensionFileSourceCode,
+  pub source_map: Option<Vec<u8>>,
```

Comment:
We should use `ExtensionFileSourceCode` here instead - otherwise source maps will always end up in compiled binary even if they are snapshotted.